components/ButtonBase.qml

```diff
@@ -41,50 +42,59 @@ Rectangle {
     Connections {
         target: obj
 
+        onStateChanged: {
+            if (obj.state === 2) {
```

Comment:
The magic number means disconnected see integrations.h but It's to use an enum need to check how to do it in QML

Reply:
Or use the `stateText` property of the entity, that returns the state as string. Not as good as enum, but if enums are not available, this might be better than a number.

Reply:
Any hint how to use a enum in qml ? @martonborzak  I don't find anything in the Qt manual

Reply:
Here's a great article about it: https://qml.guide/enums-in-qt-qml/

Reply:
@martonborzak please check my new class definition